src/physlr-molecules.cc

```diff
@@ -0,0 +1,401 @@
+using indexToBarcode_t = std::unordered_map<vertex_t, std::string>;
+using vertexSet_t = std::unordered_set<vertex_t>;
+using componentToVertexSet_t = std::vector<vertexSet_t>;
+using vertexToComponent_t = std::unordered_map<vertex_t, size_t>;
+using vecVertexToComponent_t = std::vector<vertexToComponent_t>;
+
+static void
+printVersion()
+{
+	const char VERSION_MESSAGE[] =
+	    PROGRAM " (" PACKAGE_NAME ") " GIT_REVISION "\n"
+	            "Written by Johnathan Wong.\n"
+	            "\n"
+	            "Copyright 2019 Canada's Michael Smith Genome Science Centre\n";
+	std::cerr << VERSION_MESSAGE << std::endl;
+	exit(EXIT_SUCCESS);
+}
+
+static void
+printErrorMsg(const std::string& progname, const std::string& msg)
+{
+	std::cerr << progname << ": " << msg
+	          << "\nTry 'physlr-molecules --help' for more information.\n";
+}
+
+static void
+printUsage(const std::string& progname)
+{
+	std::cout << "Usage:  " << progname
+	          << "  [-s SEPARATION-STRATEGY] [-v] FILE...\n\n"
+	             "  -v         enable verbose output\n"
+	             "  -s --separation-strategy   \n"
+	             "  SEPARATION-STRATEGY      `+` separated list of molecule separation strategies "
+	             "[bc]\n"
+	             "  --help     display this help and exit\n";
+}
+
+void
+printGraph(const graph_t& g)
+{
+	std::cout << "U\tn" << std::endl;
+	std::string node1, node2;
+	int weight;
+	auto vertexItRange = boost::vertices(g);
+	for (auto vertexIt = vertexItRange.first; vertexIt != vertexItRange.second; ++vertexIt) {
+		node1 = g[*vertexIt].name;
+		weight = g[*vertexIt].weight;
+		std::cout << node1 << "\t" << weight << "\n";
+	}
+	std::cout << "\nU\tV\tn" << std::endl;
+	auto edgeItRange = boost::edges(g);
+	for (auto edgeIt = edgeItRange.first; edgeIt != edgeItRange.second; ++edgeIt) {
+		weight = g[*edgeIt].weight;
+		node1 = g[boost::source(*edgeIt, g)].name;
+		node2 = g[boost::target(*edgeIt, g)].name;
+		std::cout << node1 << "\t" << node2 << "\t" << weight << "\n";
+	}
+}
+
+void
+readTSV(graph_t& g, const std::vector<std::string>& infiles, bool verbose)
+{
+	std::cerr << "Loading graph" << std::endl;
+#if _OPENMP
+	double sTime = omp_get_wtime();
+#endif
+
+	barcodeToIndex_t barcodeToIndex;
+	indexToBarcode_t indexToBarcode;
+	for (auto& infile : infiles) {
+		infile == "-" ? "/dev/stdin" : infile;
+		vertex_t u;
+		edge_t E;
+		std::ifstream infileStream(infile);
+		std::string line;
+		bool atEdges = false;
+		while (std::getline(infileStream, line)) {
+			if (line.empty() or line == "U\tn") {
+				continue;
+			}
+			if (line.empty() or line == "U\tV\tn") {
+				atEdges = true;
+				if (verbose) {
+					std::cerr << "Loaded vertices to graph ";
+#if _OPENMP
+					std::cerr << "in sec: " << omp_get_wtime() - sTime << std::endl;
+					sTime = omp_get_wtime();
+#endif
+				}
+				continue;
+			}
+
+			std::string node1, node2;
+			int weight;
+			std::istringstream ss(line);
+			ss >> node1;
```

Comment:
What happens if this extraction fails? Nearly every extraction needs to be wrapped in an `if`.
```suggestion
			if (ss >> node1) {
```

Reply:
Done and exit when conditional is false.